### DISCUSSION TITLE
Draw Ellipse behind resized image

### DISCUSSION THREAD

**Discussion by andyzweb** [ImageMagick/ImageMagick#2254] [Help] [answered]
I'm trying to draw a gray ellipse with some gaussian blur behind an image that has been resized and has some whitespace padding.

So far this is what I've come up with but the ellipse doesn't render in the flattened image.

`magick input.png -resize 450x450 -gravity center -extent "%[fx:w+15]"x"%[fx:h+15]" \( -clone 0 -alpha opaque -alpha extract -draw 'fill gray fill-opacity 0.5 ellipse 690,600 1000,250 0,360' \) -reverse -flatten output.png`

**Comment by snibgo** [comment]
What version of IM, on what platform?

Is input.png fully opaque? If it is, then the ellipse image will be fully obscured by the "-flatten".

Your ellipse parameters are strange. Only a small portion of the ellipse would be visible.

If that doesn't help, please link to sample images.

**Reply by andyzweb** [reply]
`magick --version
Version: ImageMagick 7.0.10-18 Q16 x86_64 2020-06-09 https://imagemagick.org
Copyright: © 1999-2020 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI Modules OpenMP(3.1) 
Delegates (built-in): bzlib freetype gslib heic jng jp2 jpeg lcms lqr ltdl lzma openexr png ps tiff webp xml zlib`

input.png is a PNG with transparency and an embedded clip path I believe.

[input.png](https://files.catbox.moe/4kxxf5.png)

yeah I still haven't drawn the ellipse in the right place :(

**Reply by andyzweb** [reply]
I guess I'm using the `-clone 0 -alpha opaque -alpha extract` to try and create a new "layer" that is the same size as the original where I can draw the ellipse. I'm not 100% positive this is the correct approach.

**Comment by snibgo** [comment, marked as answer]
See https://imagemagick.org/script/command-line-options.php#extent . "-extent" flattens against the background colour, which defaults to white, so the result is fully opaque. But you want to retain transparency, so insert "-background None" before "-extent".

Instead of "-clone 0 -alpha opaque -alpha extract", I suggest "-clone 0 -fill Black -colorize 100". It should do the same thing, but faster.

**Comment by andyzweb** [comment]
Now it looks like the blur is applying to the whole image and not just the ellipse.

`magick input.png -resize 450x450 -gravity center -background None -extent "%[fx:w+15]"x"%[fx:h+15]" \( +clone -fill Black -colorize 100 -draw 'fill gray fill-opacity 0.5 ellipse 690,600 1000,250 0,360' -blur 0x8  \) -reverse -flatten output.png`

**Comment by snibgo** [comment]
> Now it looks like the blur is applying to the whole image and not just the ellipse.

Does it? When I run that command, the gray/white boundary in the background is clearly blurred, and the product is not.

**Reply by andyzweb** [reply]
my eyes play tricks on me yet again
